bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/internal/BlueGigaSerialHandler.java

```diff
@@ -206,255 +137,80 @@ public boolean isAlive() {
         return parserThread != null && parserThread.isAlive() && !close;
     }
 
-    // Synchronize this method to ensure a packet gets sent as a block
-    private synchronized void sendFrame(BlueGigaCommand bleFrame) {
+    public void sendFrame(BlueGigaCommand bleFrame) throws IllegalStateException {
+        sendFrame(bleFrame, true);
+    }
+
+    private void sendFrame(BlueGigaCommand bleFrame, boolean checkIsAlive) throws IllegalStateException {
+        if (checkIsAlive) {
+            checkIfAlive();
+        }
+
         // Send the data
+        logger.trace("sendFrame: {}", bleFrame);
         try {
             int[] payload = bleFrame.serialize();
-            logger.trace("TX BLE frame: {}", printHex(payload, payload.length));
+            if (logger.isTraceEnabled()) {
+                logger.trace("BLE TX: {}", printHex(payload, payload.length));
+            }
             for (int b : payload) {
                 outputStream.write(b);
             }
```

Comment:
I would try to use bulk writes if possible, it would give the underlying OutputStream implementation a chance to optimize if it could. Especially if the underlying stream is unbuffered.

Reply:
I as well, but bluegiga binding use int arrays rather than byte arrays internally, which is rather cumbersome and lead to this kind "issues".